pantheon/src/test/java/tech/pegasys/pantheon/cli/BlockSubCommandTest.java

```diff
@@ -106,4 +131,62 @@ public void callingBlockImportSubCommandWithPathMustImportBlocksWithThisPath() t
     assertThat(commandOutput.toString()).isEmpty();
     assertThat(commandErrorOutput.toString()).isEmpty();
   }
+
+  // Dump sub-sub-command
+  @Test
+  public void callingBlockDumpSubCommandWithoutNumberMustDisplayErrorAndUsage() {
+    parseCommand(BLOCK_SUBCOMMAND_NAME, BLOCK_DUMP_SUBCOMMAND_NAME);
+    final String expectedErrorOutputStart = "Missing required option '--number=<LONG>'";
+    assertThat(commandOutput.toString()).isEmpty();
+    assertThat(commandErrorOutput.toString()).startsWith(expectedErrorOutputStart);
+  }
+
+  @Test
+  public void callingBlockDumpSubCommandHelpMustDisplayUsage() {
+    parseCommand(BLOCK_SUBCOMMAND_NAME, BLOCK_DUMP_SUBCOMMAND_NAME, "--help");
+    assertThat(commandOutput.toString()).startsWith(EXPECTED_BLOCK_DUMP_USAGE);
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  @Test
+  public void callingBlockDumpSubCommandWithNumberMustDisplayBlockInformation() throws Exception {
+    final Block block = generateBlock();
+    when(mockMutableBlockchain.getBlockHashByNumber(any(Long.class)))
+        .thenReturn(Optional.of(block.getHash()));
+    when(mockMutableBlockchain.getBlockByHash(any(Hash.class))).thenReturn(block);
+    parseCommand(BLOCK_SUBCOMMAND_NAME, BLOCK_DUMP_SUBCOMMAND_NAME, "--number=0");
+
+    assertThat(commandOutput.toString()).contains(block.toString());
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  @Test(expected = CommandLine.ExecutionException.class)
+  public void callingBlockDumpSubCommandWithInvalidNumberMustDisplayErrorMessage()
+      throws Exception {
+    parseCommand(BLOCK_SUBCOMMAND_NAME, BLOCK_DUMP_SUBCOMMAND_NAME, "--number=0");
+  }
+
+  @Test
+  public void callingDumpSubCommandWithFilePathMustWriteBlockInThisFile() throws Exception {
+
+    final Block block = generateBlock();
+    when(mockMutableBlockchain.getBlockHashByNumber(any(Long.class)))
+        .thenReturn(Optional.of(block.getHash()));
+    when(mockMutableBlockchain.getBlockByHash(any(Hash.class))).thenReturn(block);
+
+    final File file = File.createTempFile("dump", "store");
+
+    parseCommand(
+        BLOCK_SUBCOMMAND_NAME, BLOCK_DUMP_SUBCOMMAND_NAME, "--number=0", "--to=" + file.getPath());
+
+    assertThat(contentOf(file)).contains(block.toString());
+
+    assertThat(commandOutput.toString()).isEmpty();
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  private Block generateBlock() {
```

Comment:
There is a test util to create test blocks.  Here it is in action: https://github.com/PegaSysEng/pantheon/blob/c23fbc86f576baa2a61a4558c624ee9f8fa95c86/pantheon/src/test/java/tech/pegasys/pantheon/util/BlockImporterTest.java#L51

Reply:
Ok thanks I will update my test

Reply:
I just made the requested changes here https://github.com/PegaSysEng/pantheon/pull/1641